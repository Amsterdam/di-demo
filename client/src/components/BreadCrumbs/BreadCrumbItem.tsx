import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { themeColor } from '@datapunt/asc-ui';

export interface IBreadCrumbItemProps {
    className?: string;
    href?: string;
}

export const BreadCrumbItem: React.FC<IBreadCrumbItemProps> = ({ className, href, children }) => {
    const component: JSX.Element = <Item className={className}>{children}</Item>;

    if (href) {
        return <StyledLink to={href}>{component}</StyledLink>;
    }

    return component;
};

const StyledLink = styled(Link)`
    text-decoration: none;
`;

const Item = styled.span`
    color: ${themeColor('tint', 'level5')};

    text-decoration-color: ${themeColor('tint', 'level5')};

    &:hover {
        text-decoration: underline;
    }
`;
