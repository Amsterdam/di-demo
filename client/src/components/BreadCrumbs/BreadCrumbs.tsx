import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import styled from 'styled-components';
import { Icon, themeColor, themeSpacing, breakpoint, Theme } from '@datapunt/asc-ui';
import { ChevronRight } from '@datapunt/asc-assets';
import { BreadCrumbItem, IBreadCrumbItemProps } from './BreadCrumbItem';

interface IBreadCrumbsProps {
    className?: string;
}

interface IBreadCrumbsComposition {
    Item: React.FC<IBreadCrumbItemProps>;
}

const BreadCrumbs: React.FC<IBreadCrumbsProps> & IBreadCrumbsComposition = ({ children, className }) => {
    const themeContext = { theme: useContext(ThemeContext) as Theme.ThemeInterface };

    return (
        <Container className={className}>
            {React.Children.map(children, (child, index) => (
                <React.Fragment key={String(index)}>
                    {child}
                    {children instanceof Array && index < children.length - 1 && (
                        <StyledIcon size={14} color={themeColor('tint', 'level5')(themeContext)}>
                            <ChevronRight />
                        </StyledIcon>
                    )}
                </React.Fragment>
            ))}
        </Container>
    );
};

const Container = styled.div`
    display: flex;
    align-items: center;
    margin: ${themeSpacing(4)} 0;

    @media ${breakpoint('min-width', 'laptopM')} {
        margin: ${themeSpacing(5)} 0;
    }
`;

const StyledIcon = styled(Icon)`
    margin: 0 ${themeSpacing(1)};
`;

BreadCrumbs.Item = BreadCrumbItem;

export default BreadCrumbs;
