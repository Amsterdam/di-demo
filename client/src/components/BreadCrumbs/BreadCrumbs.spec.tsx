import React from 'react';
import { screen, act, fireEvent } from '@testing-library/react';
import { wrappedRender } from '@test/utils';
import BreadCrumbs from '@components/BreadCrumbs/BreadCrumbs';

describe('BreadCrumbs', () => {
    it('should render 3 items correctly', async () => {
        wrappedRender(
            <BreadCrumbs>
                <BreadCrumbs.Item href="/">home</BreadCrumbs.Item>
                <BreadCrumbs.Item href="/innovatie">Innovatie</BreadCrumbs.Item>
                <BreadCrumbs.Item href="/innovatie/demo">Demo</BreadCrumbs.Item>
            </BreadCrumbs>
        );
        const el: HTMLElement = await screen.findByTestId('breadCrumbs');
        expect(el).toMatchSnapshot();
    });
});
