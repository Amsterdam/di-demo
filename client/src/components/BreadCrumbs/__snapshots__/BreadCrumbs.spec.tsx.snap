// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs should render 3 items correctly 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c2 > svg {
  width: 14px;
  height: 14px;
}

.c2 svg circle,
.c2 svg rect,
.c2 svg polygon,
.c2 svg path {
  fill: #767676;
}

.c1 a {
  color: #767676;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration-color: #767676;
  text-decoration-color: #767676;
}

.c1 a:hover {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 a *:focus {
  outline: auto;
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: 2px;
  outline-offset: 4px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 16px 0;
}

.c3 {
  margin: 0 4px;
}

@media (min-width:1200px) {
  .c0 {
    margin: 20px 0;
  }
}

<nav
  aria-label="Breadcrumbs"
  class="c0"
  data-testid="breadCrumbs"
>
  <a
    href="/"
  >
    <span
      class="c1"
    >
      home
    </span>
  </a>
  <span
    class="c2 c3"
    color="#767676"
    size="14"
  >
    <svg
      focusable="false"
      viewBox="0 0 32 32"
    >
      <path
        d="M9.757 32l-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
        fill-rule="evenodd"
      />
    </svg>
  </span>
  <a
    href="/innovatie"
  >
    <span
      class="c1"
    >
      Innovatie
    </span>
  </a>
  <span
    class="c2 c3"
    color="#767676"
    size="14"
  >
    <svg
      focusable="false"
      viewBox="0 0 32 32"
    >
      <path
        d="M9.757 32l-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
        fill-rule="evenodd"
      />
    </svg>
  </span>
  <a
    href="/innovatie/demo"
  >
    <span
      class="c1"
    >
      Demo
    </span>
  </a>
</nav>
`;
